lib/utilities/addon-calculate-cache-key-for-tree.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+var crypto = require('crypto');
+
+module.exports = function calculateCacheKeyForTree(treeType, addonInstance) {
+  var cacheKeyParts = [
+    addonInstance.pkg,
```

Comment:
This seems a bit overkill, but easier than addressing all the things which could possibly invalidate.